src/txadmission.cpp

```diff
@@ -356,6 +355,12 @@ void ThreadTxAdmission()
                         LOG(MEMPOOL, "rejected orphan as likely contains old sighash");
                     }
                 }
+                else  // If the problem wasn't that the tx is an orphan, then uncache the inputs since we likely won't
+                      // need them again.
+                {
+                    for (const COutPoint &remove : vCoinsToUncache)
+                        pcoinsTip->Uncache(remove);
+                }
```

Comment:
this is actually already handled on L412 txadmission.cpp. (in accepttomemorypool)

Reply:
AcceptToMemoryPool is now a top-level API which accepts a tx in the old serialized way.  It will (for example) return whether the tx was accepted or not.  

ThreadTxAdmission is the parallel version.  So the code needs to be in both places.  In a future PR, I want to change up AcceptToMemoryPool to use the parallel code, but wait for the result of the particular tx it is interested in.  This will involve putting extra data in CTxInputData and CTxCommitData so once the tx is processed it knows to notify some other thread.